pkg/caching/caching.go

```diff
@@ -0,0 +1,48 @@
+package caching
+
+import (
+	"time"
+
+	"google.golang.org/protobuf/proto"
+
+	cache "github.com/patrickmn/go-cache"
+	"github.com/rancher/opni/pkg/storage"
+)
+
+type InMemoryEntityCache struct {
```

Comment:
This probably needs a memory limit as well.

Reply:
using ccache LRU cache with memory limits & eviction policies now